lib/hanami/model/sql/consoles/mysql.rb

```diff
@@ -0,0 +1,65 @@
+require 'shellwords'
+
+module Hanami
+  module Model
+    module Sql
+      module Consoles
+        # MySQL adapter
+        #
+        # @since x.x.x
+        # @api private
+        class Mysql
+          # @since x.x.x
+          # @api private
+          def initialize(uri)
+            @uri = uri
+          end
+
+          # @since x.x.x
+          # @api private
+          def connection_string
+            str = 'mysql'
+            str << host
+            str << database
+            str << port if port
+            str << username if username
+            str << password if password
+            str
```

Comment:
string concatenation feels kind of strange to me, what about:
```
[
  'mysql',
  host,
  database,
  port,
  username,
  password
].join
```
or on a single line even